apps/site/src/app/apply/sections/Components/DropdownSelect.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { useState } from "react";
+import RequiredAsterisk from "./RequiredAsterisk";
+
+interface SelectProps {
+	labelStyle: string;
+	inputStyle: string;
+	name: string;
+	labelText: string;
+	values: Array<{ value: string; text: string }>;
+	containerClass: string;
+}
+
+interface OtherProps {
+	value: string;
+	name: string;
+}
+
+const OtherPopup = ({ value, name }: OtherProps) => {
+	if (value == "other") {
+		return (
+			<div className="mt-2 flex gap-2">
+				<label htmlFor={`${name}-other-input`} className="text-lg">
+					Other: <RequiredAsterisk />
+				</label>
+				<input
+					type="text"
+					name={`${name}-other-input`}
+					id={`${name}-other-input`}
+					className="border-b-2 p-1 h-6 border-black w-6/12"
+					required
+				/>
+			</div>
+		);
+	}
+};
+
+export default function DropdownSelect({
+	labelStyle,
+	inputStyle,
+	name,
+	labelText,
+	values,
+	containerClass,
+}: SelectProps) {
+	const [value, setValue] = useState("");
+
+	return (
+		<div className={containerClass}>
+			<label className={`${labelStyle}`} htmlFor={name}>
+				{labelText} <RequiredAsterisk />
+			</label>
+			<select
```

Comment:
Issue: This `select` field needs to be accessible, meaning clicking the label should focus the user's cursor on the `select` field. This can be done by setting the `htmlFor` of the `label` to the `id` of the `select` field, not the `name`.